### PR TITLE
Updates to the function to automatically relax Pyomo models

### PR DESCRIPTION
**Pull request by michaelbynum**
1. Allow users to pass options to the call to `fbbt`
2. Add a docstring to `relax`
3. Add some notes on the reasoning for using FBBT both before and after relaxing the model. Also added some notes on how to correct this behavior in the future.

### CODE REVIEW

File: coramin/relaxations/auto_relax.py [hunk 1]
```diff
@@ -857,12 +857,57 @@ def _relax_expr(expr, aux_var_map, parent_block, relaxation_side_map, counter, d
...
+    model: pyomo.core.base.block._BlockData or pyomo.core.base.PyomoModel.ConcreteModel
+        The model or block to be relaxed
+    descend_into: type or tuple of type
+        The types of pyomo components that should be checked for constraints to be relaxed. The
+        default is (Block, Disjunct).
+    in_place: bool
+        If True, then model will be modified in place. If False, model will be cloned, and
+        the clone will be relaxed.
+    use_fbbt: bool
+        If True, then FBBT will be used to tighten variable bounds.
```
Comment (jsiirola) [inline coramin/relaxations/auto_relax.py]: Can you add the default to this docstring (reading it implies that the default is False, which is not correct).

File: coramin/relaxations/auto_relax.py [hunk 2]
```diff
@@ -857,12 +857,57 @@ def _relax_expr(expr, aux_var_map, parent_block, relaxation_side_map, counter, d
...
+    fbbt_options: dict
+        The options to pass to the call to fbbt. See pyomo.contrib.fbbt.fbbt.fbbt for details.
```
Comment (jsiirola) [inline coramin/relaxations/auto_relax.py:832]: Is there a way to make the reference to `pyomo.contrib.fbbt.fbbt.fbbt` a cross-link to the pyomo documentation?  Or should we wait until there is readthedocs for coramin?

File: coramin/relaxations/auto_relax.py [hunk 3]
```diff
@@ -857,12 +857,57 @@ def _relax_expr(expr, aux_var_map, parent_block, relaxation_side_map, counter, d
...
 
 
-def relax(model, descend_into=None, in_place=False, use_fbbt=True):
+def relax(model, descend_into=None, in_place=False, use_fbbt=True, fbbt_options=None):
+    """
+    Create a convex relaxation of the model.
+
+    Parameters
+    ----------
+    model: pyomo.core.base.block._BlockData or pyomo.core.base.PyomoModel.ConcreteModel
```
Comment (jsiirola) [inline coramin/relaxations/auto_relax.py:820]: Can you make crosslinks to the pyomo documentation?

'

File: coramin/relaxations/auto_relax.py [hunk 4]
```diff
@@ -857,12 +857,57 @@ def _relax_expr(expr, aux_var_map, parent_block, relaxation_side_map, counter, d
...
+        The model or block to be relaxed
+    descend_into: type or tuple of type
+        The types of pyomo components that should be checked for constraints to be relaxed. The
+        default is (Block, Disjunct).
+    in_place: bool
+        If True, then model will be modified in place. If False, model will be cloned, and
+        the clone will be relaxed.
```
Comment (jsiirola) [inline coramin/relaxations/auto_relax.py]: Can you add the default value to the docstring?

**Review by jsiirola — APPROVED** [review]
Looks good, but I have a few minor updates to the docstring (specify the default parameter value)

**Comment by michaelbynum** [comment]
@jsiirola Thanks - good suggestions. I think we should wait on the crosslinks until we have readthedocs because I don't have an easy way to ensure the link is working correctly.

**Pull request merged**